worlds/sc2/Client.py

```diff
@@ -833,6 +833,11 @@ def calculate_items(ctx: SC2Context) -> typing.Dict[SC2Race, typing.List[int]]:
         if item_data.quantity == 1:
             accumulators[item_data.race][item_data.type.flag_word] |= 1 << item_data.number
 
+            # Guardian Shell breaks without SoA on version 4+, but can be generated without SoA on version 3
+            if ctx.slot_data_version < 4 and name == ItemNames.GUARDIAN_SHELL \
+                    and ctx.spear_of_adun_autonomously_cast_ability_presence != SpearOfAdunAutonomouslyCastAbilityPresence.option_everywhere:
+                ctx.spear_of_adun_autonomously_cast_ability_presence = SpearOfAdunAutonomouslyCastAbilityPresence.option_protoss
+
```

Comment:
This won't do actually anything as options are set first

Reply:
I think the simplest way would be adding an if to the mentioned block that checks version and the option

Reply:
I think what you're suggesting would be a lot more disruptive than doing nothing at all, though. Doing nothing means sometimes Guardian Shell breaks on older games; setting autocast to "protoss" means players are potentially forced to play with Reconstruction Beam and Overwatch in prophecy/prologue, with no way to opt-out, even if their game doesn't have Guardian Shell.

Reply:
Applied the requested change anyway though